### PR TITLE
CDK: Add OAuth and Streams/Core tests.

### PR DESCRIPTION
**Pull request by davinchia**
## What
Completely test the `OAuth` module and almost all of the `streams/core.py` file.

Add `pytest-mock` as well as a convenience wrapper to run the type checker and the test together when developing.

Also configure `pytest` to pick up loggers at the INFO level and above, and pick up standard out for easier debugging.

I do need to write test for `streams/core.py`'s `get_json_schema` will follow up tomorrow.

I know some of the script bits are unpolished - I will improve them as we go along. 

## How
As above.

## Recommended reading order
1. `test_auth.py`
1. `test_streams_core.py`

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/base_python/test_schema_helpers.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import os
 import shutil
 import sys
-from collections import Mapping
+from collections.abc import Mapping
```
Comment (davinchia) [inline airbyte-cdk/python/unit_tests/base_python/test_schema_helpers.py:28]: this was throwing an warning - importing from just `collections` is going to be deprecated with Python 3.9.

**Comment by davinchia** [comment]
This brings us to 38%!

Screen Shot 2021-05-05 at 9 43 55 PM

File: airbyte-cdk/python/.coveragerc [hunk 2]
```diff
@@ -1,4 +1,11 @@
+[report]
+# show lines missing coverage
+show_missing = true
```
Comment (sherifnada) [inline airbyte-cdk/python/.coveragerc:3]: wouldn't this make cov very noisy?

File: airbyte-cdk/python/unit_tests/base_python/test_auth.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            TestOauth2Authenticator.refresh_endpoint,
+            TestOauth2Authenticator.client_id,
+            TestOauth2Authenticator.client_secret,
+            TestOauth2Authenticator.refresh_token,
+        )
+
+        expire_immediately = 0
+        mocker.patch.object(Oauth2Authenticator, "refresh_access_token", return_value=("access_token_1", expire_immediately))
+        oauth.get_auth_header()  # Set the first expired token.
+
```
Comment (sherifnada) [inline airbyte-cdk/python/unit_tests/base_python/test_auth.py:94]: should we have an assertion here too?

File: airbyte-cdk/python/unit_tests/base_python/test_streams_core.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+# SOFTWARE.
+
+
+from typing import Any, Iterable, List, Mapping, Union
+
+from airbyte_cdk.base_python import Stream
+from airbyte_cdk.models import AirbyteStream, SyncMode
+
+
+class StreamStubFullRefresh(Stream):
```
Comment (sherifnada) [inline airbyte-cdk/python/unit_tests/base_python/test_streams_core.py:30]: there's probably more testing we can do for shaping the `AirbyteStream` based on: 

* primary key
* `supports_incremental` returns true if cursor field is set

but LGTM

**Review by sherifnada — APPROVED** [review]

**Pull request merged**